pytorch_lightning/loggers/base.py

```diff
@@ -221,13 +221,19 @@ def version(self) -> Union[int, str]:
 class LoggerCollection(LightningLoggerBase):
     """The :class:`LoggerCollection` class is used to iterate all logging actions over the given `logger_iterable`.
 
+    .. deprecated:: v1.6
+        `LoggerCollection` is deprecated in v1.6 in favor of `trainer.loggers` and will be removed in v1.8.
```

Comment:
I would prefer to make the error message more explicit. The short changelog entry is fine but here we need to say what the user should do instead. 

```suggestion
        `LoggerCollection` is deprecated in v1.6 and will be removed in v1.8. Directly pass a list of loggers to the Trainer and access the list via the `trainer.loggers` attribute. 
```

Reply:
I think this is fine, but the user actually doesn't have to change any behavior, they still pass a list of loggers to the Trainer

Reply:
But hold on, this warning is when LoggerCollection gets instantiated by the user. This is what we want to warn about, and passing it to the trainer should trigger a warning as well. However, we need to avoid triggering the warning internally or multiple times. This is currently  a problem, because for the getter in 

`trainer.logger`

we return 

`return LoggerCollection(self.loggers)`

which will once again trigger a warning that the user will find confusing and can't change. Do you agree?
Can we make an internal flag that we flip when we want to suppress the warning internally?

Reply:
We should still warn about that scenario because the behavior will change following v1.8 as users will receive the first logger in the list instead of a logger collection. To update their code, they would need to use trainer.loggers

Reply:
@awaelchli after https://github.com/PyTorchLightning/pytorch-lightning/pull/11920 there are virtually no places where we call `trainer.logger` internally - we updated them all to `trainer.loggers`

Reply:
@daniellepintz To be very clear:

```py
from pytorch_lightning import Trainer
from pytorch_lightning.loggers import TensorBoardLogger


if __name__ == "__main__":
    trainer = Trainer(
        logger=[TensorBoardLogger(save_dir="."), TensorBoardLogger(save_dir=".")]
    )
    _ = trainer.logger
```

Nothing in the above code is deprecated usage. Yet we see this message:

```
/Users/adrian/repositories/pytorch-lightning/pytorch_lightning/trainer/trainer.py:2621: UserWarning: Using trainer.logger when Trainer is configured to use multiple loggers. This behavior will change in v1.8 when LoggerCollection is removed, and trainer.logger will return the first logger in trainer.loggers
  rank_zero_warn(
/Users/adrian/repositories/pytorch-lightning/pytorch_lightning/loggers/base.py:235: LightningDeprecationWarning: `LoggerCollection` is deprecated in v1.6 and will be removed in v1.8. Directly pass a list of loggers to the Trainer and access the list via the `trainer.loggers` attribute.
  rank_zero_deprecation(
```

There is nothing I as a user can do to avoid the second warning, except change to `trainer.loggers` but that's not deprecated API. Plus, the warning mentions `LoggerCollection` which I have not used anywhere in my code. 

It's triggered internally because we instantiate `LoggerCollection` and we need to suppress that warning, for example with https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings or the toggle as I showed above. No matter how unlikely it is that a user runs into this issue, we can't do it because it's not correct.

Reply:
We were actually having a similar discussion earlier. https://github.com/PyTorchLightning/pytorch-lightning/pull/12147#r816308356. Having the user see both the warning and the deprecation was an intentional choice (originally both messages were combined into one). I agree though that it would be confusing for the user to see the deprecation warning from instantiating `LoggerCollection` so this is a good solution